tutorials/tracing.md

```diff
@@ -0,0 +1,189 @@
+# CRI-O Tracing
+
+## Configuration
+
+To enable [OpenTelemetry][otel] tracing support in CRI-O, either start `crio`
+with `--enable-tracing` or add the corresponding option to a config overwrite,
+for example `/etc/crio/crio.conf.d/01-tracing.conf`:
+
+[otel]: https://opentelemetry.io
+
+```toml
+[crio.tracing]
+enable_tracing = true
+```
+
+Traces in CRI-O get exported via the [OpenTelemetry Protocol][otlp] by using an
+[gRPC][grpc] endpoint. This endpoint defaults to `0.0.0.0:4317`, but can be
+configured by using the `--tracing-endpoint` flag or the corresponding TOML
+configuration:
+
+[otlp]: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md
+[grpc]: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md#otlpgrpc
+
+```toml
+[crio.tracing]
+tracing_endpoint = "0.0.0.0:4317"
+```
+
+The final configuration aspect of OpenTelemetry tracing in CRI-O is the
+`--tracing-sampling-rate-per-million` / `tracing_sampling_rate_per_million`
+configuration, which refers to the amount of samples collected per million
+spans. This means if it being set to `0` (the default), then CRI-O will not
+collect any traces at all. If set to `1000000` (one million), then CRI-O will
+create traces for all created spans. If the value is below one million, then
+there is **no** way right now to select a subset of spans other than the modulo
+of the set value.
+
+[conmon-rs][conmon-rs] has the capability to add additional tracing past the
+scope of CRI-O. This is automatically enabled when the `pod` runtime type is
+chosen, like so:
+
+```toml
+[crio.runtime]
+default_runtime = "runc"
+
+[crio.runtime.runtimes.runc]
+runtime_type = "pod"
+```
+
+[conmon-rs]: https://github.com/containers/conmon-rs
+
+Then conmon-rs will export traces and spans in the same way CRI-O does
+automatically. Both CRI-O and conmon-rs will correlate their logs to the traces
+and spans. If the connection to the OTLP instance gets lost, then CRI-O will not
+block, but all the traces during that time will be lost.
+
+## Usage example
+
+The [OpenTelemetry Collector][collector] alone cannot be used to visualize
+traces and spans. For that a frontend like [Jaeger][jaeger] can be used to
+connect to it. To achieve that, a configuration file for OTLP needs to be
+created, like this `otel-collector-config.yaml`:
+
+[jaeger]: https://www.jaegertracing.io
+[collector]: https://opentelemetry.io/docs/collector
+
+```yaml
+receivers:
+  otlp:
+    protocols:
+      http:
+      grpc:
+
+exporters:
+  logging:
+    loglevel: debug
+
+  jaeger:
+    endpoint: localhost:14250
+    tls:
+      insecure: true
+
+processors:
+  batch:
+
+extensions:
+  health_check:
+  pprof:
+    endpoint: :1888
```

Comment:
Looks like `localhost` is the default if not specified, I recommend adding for consistency with previous entries. i.e.  `endpoint: localhost:1888`

Reply:
Fixed  as suggested :+1: